sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Common/src/Shared/BinaryDataExtensions.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Storage.Common
+{
+    internal static class BinaryDataExtensions
+    {
+        private static readonly UTF8Encoding encoding = new UTF8Encoding(false, true);
+
+        public static string ToValidUTF8String(this BinaryData binaryData)
+        {
+            if (MemoryMarshal.TryGetArray(binaryData.ToMemory(), out ArraySegment<byte> data))
+            {
+                return encoding.GetString(data.Array, data.Offset, data.Count);
+            }
+            return encoding.GetString(binaryData.ToArray());
+        }
+
+        public static string TryGetAsString(this BinaryData binaryData)
+        {
+            if (binaryData == null)
+            {
+                throw new ArgumentNullException(nameof(binaryData));
+            }
+
+            string value;
+
+            try
+            {
+                value = binaryData.ToValidUTF8String();
+            }
+            catch (Exception ex)
+            {
+                if (ex is DecoderFallbackException || ex is FormatException)
```

Comment:
Was this the behavior of Track 1 based SDK?

Reply:
This comes from webjobs actually https://github.com/Azure/azure-webjobs-sdk/blob/4130350327c6d637d48456222de7e658c6cf729a/src/Microsoft.Azure.WebJobs.Extensions.Storage/Queues/StorageQueueMessageExtensions.cs .

And was based on track 1 behavior https://github.com/Azure/azure-storage-net/blob/4bd9460e0b9b007bdb34eaa1329de2257f7a34dc/Lib/Common/Queue/CloudQueueMessage.Common.cs#L86 .

I also logged https://github.com/Azure/azure-sdk-for-net/issues/16951 for BinaryData to offer similar support.

There's plenty of tests that depend on this behavior.

Reply:
Maybe adding a log might be nice.